### PR TITLE
Adding Milestones capabilities to Linear extension

### PR DESCRIPTION
**Pull request by mil3na**
## Description

Adding milestone functionality to the Linear extension, close https://github.com/raycast/extensions/issues/6388

## Screencast

(a video can be provided if needed, but I found no way to share this without sharing confidential info)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/linear/src/api/getMilestones.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  description
+  targetDate
+  archivedAt
+  createdAt
+  project {
+      id
+  }
+  sortOrder
+  updatedAt
+`;
```
Comment (thomaslombart) [inline extensions/linear/src/api/getMilestones.ts:26]: Are all milestones properties useful here? As far as it looks, `archivedAt`, `createdAt`, and `description` may be useless.

File: extensions/linear/src/components/IssueActions/MilestoneSubmenu.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      successTitle: milestone ? "Set milestone" : `Removed milestone from ${issue.identifier}`,
+      successMessage: milestone ? `"${milestone.name}" added to ${issue.identifier}` : "",
+      errorTitle: "Failed to set milestone",
+    });
+  }
+
+  return (
+    <ActionPanel.Submenu
+      title="Set Milestone"
+      icon={{ source: "linear-icons/milestone.svg", tintColor: Color.PrimaryText }}
```
Comment (thomaslombart) [inline extensions/linear/src/components/IssueActions/MilestoneSubmenu.tsx:46]: `Color.PrimaryText` is already the default color.

```suggestion
      icon="linear-icons/milestone.svg"
```

File: extensions/linear/src/components/ProjectIssues.tsx [hunk 3]
```diff
@@ -18,9 +22,54 @@ type ProjectIssuesProps = {
 
 export default function ProjectIssues({ projectId, priorities, me, users }: ProjectIssuesProps) {
   const { issues, isLoadingIssues, mutateList } = useIssues(getProjectIssues, [projectId]);
+  const [milestone, setMilestone] = useCachedState<string>("");
```
Comment (thomaslombart) [inline extensions/linear/src/components/ProjectIssues.tsx:25]: Can't you use `storeValue` on `List.Dropdown` instead?

File: extensions/linear/src/components/ProjectIssues.tsx [hunk 4]
```diff
@@ -18,9 +22,54 @@ type ProjectIssuesProps = {
...
+  }, [milestone, issues, milestones]);
 
   return (
-    <List isLoading={isLoadingIssues} searchBarPlaceholder="Filter by ID, title, status, assignee or priority">
+    <List
+      isLoading={isLoadingIssues}
+      {...(milestones && milestones.length > 0
+        ? {
+            searchBarAccessory: (
+              <List.Dropdown tooltip="Change Roadmap" onChange={setMilestone} value={milestone}>
```
Comment (thomaslombart) [inline extensions/linear/src/components/ProjectIssues.tsx:53]: Shouldn't it be `Change Milestone` instead?

```suggestion
              <List.Dropdown tooltip="Change Milestone" onChange={setMilestone} value={milestone}>
```

File: extensions/linear/src/components/CreateIssueForm.tsx [hunk 5]
```diff
@@ -198,6 +205,7 @@ export default function CreateIssueForm(props: CreateIssueFormProps) {
   const { cycles } = useCycles(values.teamId);
   const { issues } = useIssues(getLastCreatedIssues);
   const { projects } = useProjects(values.teamId);
+  const { milestones } = useMilestones(values.projectId);
```
Comment (thomaslombart) [inline extensions/linear/src/components/CreateIssueForm.tsx:208]: Let's use the `execute` boolean to make sure milestones are loaded only once a project is selected.

```suggestion
  const { milestones } = useMilestones(values.projectId, { execute: !!values.projectId} );
```

**Review by thomaslombart — COMMENTED** [review]
Awesome work!

FYI, you can uses the word "close" before the issue number in the description so GitHub automatically closes it once the PR is merged.

File: extensions/linear/src/api/getMilestones.ts (hunk 1 above)
Comment (mil3na) [reply extensions/linear/src/api/getMilestones.ts:26]: That's a good point! I will remove it for now.

File: extensions/linear/src/components/ProjectIssues.tsx (hunk 4 above)
Comment (mil3na) [reply extensions/linear/src/components/ProjectIssues.tsx:53]: Yes! Good catch!

File: extensions/linear/src/components/ProjectIssues.tsx (hunk 3 above)
Comment (mil3na) [reply extensions/linear/src/components/ProjectIssues.tsx:25]: Good question. I can try to make it work. Would that make a big difference?

Comment (mil3na) [reply extensions/linear/src/components/ProjectIssues.tsx:25]: The reason why I'm using it on the cache is so I can use for filtering the list.

Comment (thomaslombart) [reply extensions/linear/src/components/ProjectIssues.tsx:25]: It doesn't make a big difference, it's just a bit more maintainable.

**Review by thomaslombart — COMMENTED** [review]
I couldn't find any reference to `milestone-complete` in the code? Is it used somewhere?

**Comment by mil3na** [comment]
> I couldn't find any reference to `milestone-complete.svg` in the code? Is it used somewhere?

Not yet, as we don't have easy access to this info yet. I thought about uploading it so it doesn't get lost. Shall I remove it?

**Comment by thomaslombart** [comment]
No, it's fine! Just wanted to double-check.

**Review by thomaslombart — APPROVED** [review]

**Pull request merged**